From 04712062cef4f1154503417070ac50cf4b36472b Mon Sep 17 00:00:00 2001
From: Fancy Fang <B47543@freescale.com>
Date: Wed, 8 Jan 2014 18:06:57 +0800
Subject: [PATCH 254/259] ENGR00294210 G2D: fix build errors in Yocto system

The G2D's compiling system is different from the Yocto's.
So the previous Makefile cannot be used to compile g2d lib
in Yocto environment.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 g2d/Makefile | 63 +++++++++++++++++++++++++++++-------------------------------
 1 file changed, 30 insertions(+), 33 deletions(-)

diff --git a/g2d/Makefile b/g2d/Makefile
index 73074fd..3118841 100644
--- a/g2d/Makefile
+++ b/g2d/Makefile
@@ -1,44 +1,41 @@
-#/*
-# *  Copyright (C) 2013-2014 Freescale Semiconductor, Inc.
-# *  All Rights Reserved.
-# *
-# *  The following programs are the sole property of Freescale Semiconductor Inc.,
-# *  and contain its proprietary and confidential information.
-# *
-# */
-#
-# Linux build file for g2d library
-#
-.PHONY: clean
+CC ?=$(CROSS_COMPILE)gcc
+AR ?=$(CROSS_COMPILE)ar
+
+# list of platforms which want this test case
+INCLUDE_LIST:=IMX50 IMX51 IMX5 IMX6Q IMX6S
+
+OBJ = g2d.o
 
-AQROOT ?= ../../../driver/
+LIBNAME = libg2d
+SONAMEVERSION = 0.7
+LFLAGS += -lpthread
 
-include $(AQROOT)/makefile.linux.def
+ifeq ($(PLATFORM), $(findstring $(PLATFORM), $(INCLUDE_LIST)))
 
-OBJS := g2d.o
+all: $(LIBNAME).so
 
-# build as shared library
-TARGET := libg2d.so
+install: install_headers
+	@mkdir -p $(DEST_DIR)/usr/lib
+	cp -P $(LIBNAME).* $(DEST_DIR)/usr/lib
 
-CFLAGS += -I ../include -fPIC -I $(AQROOT)/sdk/inc -I$(ROOTFS_USR)/include
-ifeq ($(SDK_BUILD), 1)
-LFLAGS = -L $(SDK_DIR)/drivers -L $(ROOTFS_USR)/lib -lpxp -lpthread
+install_headers:
+	@mkdir -p $(DEST_DIR)/usr/include
 else
-LFLAGS = -L $(ROOTFS_USR)/lib -lpxp -lpthread
+all install :
 endif
 
-CC = $(CROSS_COMPILE)gcc
+%.o: %.c
+	$(CC) -D$(PLATFORM) $(INCLUDE) -Wall -O2 -fPIC -c $^ -o $@
 
-$(TARGET): $(OBJS)
-	$(CC) -shared -fPIC -Wl,-soname,$@ -o $@ $(OBJS) $(LFLAGS)
-ifeq ($(SDK_BUILD), 1)
-	-@cp $@ $(SDK_DIR)/drivers/
-endif
-	-@mkdir -p ../lib/
-	-@cp $@ ../lib/
+$(LIBNAME)-pxp.so: $(OBJ)
+	$(CC) -shared -nostartfiles -Wl,-soname,$(LIBNAME).so.$(SONAMEVERSION) $(LFLAGS) $^ -o $@
 
-clean:
-	-@rm $(TARGET) $(OBJS)
+$(LIBNAME).so: $(LIBNAME).so.$(SONAMEVERSION)
+	ln -s $< $@
 
-%.o: %.c
-	$(CC) $(CFLAGS) -o $@ -c $<
+$(LIBNAME).so.$(SONAMEVERSION): $(LIBNAME)-pxp.so
+	ln -s $< $@
+
+.PHONY: clean
+clean:
+	rm -f $(LIBNAME)* $(OBJ)
-- 
1.8.0

